components/playlist/browser/media_detector_component_manager.cc

```diff
@@ -19,8 +20,38 @@ namespace playlist {
 
 namespace {
 
-base::FilePath GetScriptPath(const base::FilePath& install_path) {
-  return install_path.AppendASCII("index.js");
+using ScriptName = base::FilePath::StringType;
+using ScriptToSchemefulSiteMap = base::flat_map<ScriptName, net::SchemefulSite>;
+using ScriptToResourceIdMap = base::flat_map<ScriptName, int>;
+
+const base::FilePath::StringType& GetBaseScriptName() {
+  static const base::NoDestructor base_script(
+      ScriptName(FILE_PATH_LITERAL("index.js")));
+  return *base_script;
+}
+
+const ScriptToSchemefulSiteMap& GetScriptNameToSchemefulSiteMap() {
+  static const base::NoDestructor script_name_to_schemeful_sites(
+      ScriptToSchemefulSiteMap{
+          {FILE_PATH_LITERAL("youtube.com.js"),
+           net::SchemefulSite(GURL("https://youtube.com"))}});
+
+  return *script_name_to_schemeful_sites;
+}
+
+const base::flat_map<ScriptName, std::string>& GetLocalScriptMap() {
+  static const base::NoDestructor local_resource_map(([]() {
+    const auto& rb = ui::ResourceBundle::GetSharedInstance();
+    base::flat_map<ScriptName, std::string> local_resource_map = {
+        {GetBaseScriptName(),
+         rb.LoadDataResourceString(IDR_PLAYLIST_MEDIA_DETECTOR_JS)},
+        {FILE_PATH_LITERAL("youtube.com.js"),
+         rb.LoadDataResourceString(IDR_PLAYLIST_MEDIA_DETECTOR_YOUTUBE_JS)},
```

Comment:
cc @atuchin-m

Reply:
Please remove caching here.
See https://bravesoftware.slack.com/archives/C7VLGSR55/p1675100805299529

Reply:
Great! Replaced with RawResource version and removed caching

Reply:
Oops, calling RawResource doesn't work. I thought the script were not zipped but I seem to be wrong. Reverted to LoadDataResourceString.

Reply:
Ok. Removing the cache in enough here.